src/types/query.d.ts

```diff
@@ -253,6 +434,10 @@ export module query {
   export function ToMillis(expr: ExprArg): Expr
   export function ToMicros(expr: ExprArg): Expr
 
+  // TODO
   export function MoveDatabase(from: ExprArg, to: ExprArg): Expr
-  export function Documents(collection: ExprArg): Expr
+
+  export function Documents<T extends object>(
+    collection: Expr.CollectionRef<T>
+  ): Expr.SetRef<Document<T>>
```

Comment:
`Documents()` function won't return a set of `Document`, but instead will return a set of `Ref`s

Reply:
that's true, but the `Expr.SetRef` type makes declaring such more convenient by wrapping its type parameter with `Expr.Ref`.

so that means `Expr.SetRef<Document<T>>` resolves to `Expr.Ref<Array<Expr.Ref<Document<T>>>>`

of course, that will change slightly when i update `Expr.SetRef` as suggested

Reply:
Conceptually your `SetRef` definition is not defining what a set is, so this is making functions like this giving the wrong impression to the users that `Documents(...)` will return a set of documents, but instead this will return a set of references, I would like to see the definition of that function something like:
```typescript
  export function Documents<T extends object>(
     collection: Expr.CollectionRef<T>
   ): Expr.SetRef<DocumentRef<T>>
```
then update the `SetRef` accordingly

Reply:
Doesn't a `SetRef` always contain `Ref` values only?

Reply:
Nope. If you page an index with a non-default values set you'll get a set over those values, which are not constrained to be refs.

Reply:
no, the `Match(...)` function is an example (which is wrong by your definition) that returns a SetRef, but when materialized by `Paginate` will transform into pages of type `T`, depending on the index definition